dynamiqs/solvers/ode/fixed_solver.py

```diff
@@ -31,14 +31,8 @@ class FixedSolver(AutogradSolver):
 
     def __init__(self, *args):
         super().__init__(*args)
+
         self.dt = self.options.dt
-        if isinstance(self.dt, Tensor):
-            if self.dt.numel() == 1:
-                self.dt = self.dt.item()
-            else:
-                raise ValueError(
-                    f'`dt` should be a number or a 0-d tensor, but is {self.dt}.'
-                )
```

Comment:
We indeed need to check that `dt` is a float, and not a Tensor. Otherwise this was creating a bug somewhere in the adjoint (forgot where) and also it respects the required typing.

`dt` is a Tensor for instance if you pass:
```python
dq.solver.Rouchon(dt=tsave[1]-tsave[0])
```

Can you add back the conversion from 0-D Tensor to float where you think it's appropriate? (in the options class maybe).

Reply:
Ok, I add it back. Do we really need to check that it's a 0-D tensor with one element? It seems a bit far fetched that someone would put a 1d tensor for dt no?

I propose to add in the options
```diff
class _ODEFixedStep(Solver):
    SUPPORTED_GRADIENT = (Autograd,)

    def __init__(self, *, dt: float):
+        # convert dt because it's common to pass a 0-D tensor instead of a float
+        if isinstance(dt, Tensor):
+            dt = dt.item()
        self.dt = dt
```

Reply:
Indeed, as long as `dt.item()` works (and it should for N-D tensors as long as they have 1 element), it's good.